backend/scripts/mongo/seedQuestions.js

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+const { MongoClient, ObjectId } = require('mongodb');
+const questionsJson = require('../../data/questions.json');
+const fs = require('fs').promises;
+
+const generateFilePaths = async () => {
+  const questionDescFilePaths = [];
+
+  for (let i = 1; i <= questionsJson.length; i++) {
+    questionDescFilePaths.push(`data/question-descriptions/${i}.md`);
+  }
+
+  return questionDescFilePaths;
+};
+
+const readFiles = async () => {
+  try {
+    const questionDescFilePaths = await generateFilePaths();
+    const descriptions = [];
+
+    await Promise.all(
+      questionDescFilePaths.map(async (filePath, index) => {
+        try {
+          const data = await fs.readFile(filePath, 'utf8');
+          descriptions[index] = data;
+          console.log('Read and pushed:', filePath);
+        } catch (err) {
+          console.error(`Error reading file ${filePath}: ${err}`);
+        }
+      }),
+    );
+
+    return descriptions;
+  } catch (error) {
+    console.error('Error generating file paths:', error);
+    return [];
+  }
+};
+
+const uri = 'mongodb://127.0.0.1:27017/';
+
+async function connectToDatabase() {
+  try {
+    // Open connection to MongoDB
+    const client = new MongoClient(uri);
+    await client.connect().then(() => console.log('Connected to MongoDB'));
+
+    const database = client.db('peer-prep');
+
+    const questionCategoriesCollection =
+      database.collection('questioncategories');
+    const questionCollection = database.collection('questions');
+    const difficultyCollection = database.collection('difficulties');
+    const categoryCollection = database.collection('categories');
+
+    await questionCategoriesCollection
+      .deleteMany({})
+      .then(() => console.log('Deleted all question categories'));
+    await questionCollection
+      .deleteMany({})
+      .then(() => console.log('Deleted all questions'));
```

Comment:
Let's do an upsert instead, as per #71, to prevent accidentally deleting the existing question IDs in dev or prod (since other services depend on these question IDs and we can't enforce a fkey violation when deleting these entries)

Reply:
https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g20/pull/71/files#diff-b36838d0b610dcf5d544f4851d605687dc67a2449f0a071ceb81123a7c8f0204